tests/test_rtm.py

```diff
@@ -0,0 +1,343 @@
+from eventlet import sleep
+from eventlet.event import Event
+from mock import call, Mock, patch
+import pytest
+
+from nameko_slack import rtm
+
+
+@pytest.fixture
+def tracker():
+    yield Mock()
+
+
+@pytest.fixture
+def config():
+    return {}
+
+
+@pytest.fixture
+def service_runner(container_factory, config):
+    """
+    Service runner
+
+    Return a utility test function which runs the given service
+    and sets mocked Slack client to "publish" a given set of events
+
+    """
+
+    def _runner(service_class, events):
+
+        with patch('nameko_slack.rtm.SlackClient') as SlackClient:
+            SlackClient.return_value.rtm_read.return_value = events
+            container = container_factory(service_class, config)
+            container.start()
+            sleep(0.1)  # enough to handle all the test events
+
+    return _runner
+
+
+@pytest.fixture
+def make_message_event():
+    """ Sample message event maker
+    """
+    def _make(**overrides):
+        event = {
+            'type': 'message',
+            'user': 'U11',
+            'text': 'spam',
+            'channel': 'D11',
+            'ts': '1480798992.000002',
+            'team': 'T11',
+        }
+        event.update(overrides)
+        return event
+    return _make
+
+
+@pytest.fixture
+def events(make_message_event):
+    return [
+        {'type': 'hello'},
+        {'type': 'presence_change', 'presence': 'active', 'user': 'U11'},
+        make_message_event(text='spam ham'),
+        {'type': 'presence_change', 'presence': 'away', 'user': 'U00'},
+        make_message_event(text='ham spam'),
+        {},  # emtpy event, no type specified
+        make_message_event(text='spam egg'),
+    ]
+
+
+class TestHandleEvents:
+
+    def test_handle_any_event(self, events, service_runner, tracker):
+
+        class Service:
+
+            name = 'sample'
+
+            @rtm.handle_event
+            def handle_event(self, event):
+                tracker.handle_event(event)
+
+        service_runner(Service, events)
```

Comment:
Rather than sleeping in the `service_runner` you could use the `entrypoint_waiter` here, and wait for `handle_event` to fire the `len(events)` times.

I'm not sure that would be better, but thought I'd mention it all the same.